pkg/kubectl/cmd/apply.go

```diff
@@ -101,6 +104,7 @@ func NewCmdApply(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmdutil.AddValidateFlags(cmd)
 	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on")
 	cmd.Flags().Bool("all", false, "[-all] to select all the specified resources.")
+	cmd.Flags().StringArrayP("groupversionkind", "g", []string{}, "Overwrite the default whitelist with <group/version/kind> for --prune")
```

Comment:
We kind of try to have dashes in flag names as a convention. Given you have a short form, could this be `--group-version-kind`, or something directly related to the purpose instead of the format, like `--prune-whitelist`, `--skip-when-pruning`, `--prune-skipped`, or something similar? Starting with `--prune*` would also make them appear together in the flags list.  ;)